surfnperf.js

```diff
@@ -356,6 +356,59 @@
     return this.duration('navigationStart', 'loadEventEnd');
   };
 
+  /**
+   * Time to first paint (browser-reported time for the first pixels to be rendered on the page)
+   *
+   * @returns {integer} time in ms
+   * @memberOf SurfNPerf
+   */
+  SNPProto.getFirstPaint = function(options) {
+    if(this.chromeLoadTimes()) {
+      this._data.marks.chromePaint = this.chromeLoadTimes().firstPaintTime * 1000;
+      return this.duration('navigationStart', 'chromePaint', options);
+    } else if(this.msFirstPaint()) {
+      this._data.marks.msPaint = this.msFirstPaint();
+      return this.duration('navigationStart', 'msPaint', options);
+    } else {
+      return null;
+    }
+  };
+
+  /**
+   * Time to first paint (Time calculated based on window.requestAnimationFrame() for the first pixels to be rendered on the page)
+   *
+   * @returns {integer} time in ms
+   * @memberOf SurfNPerf
+   */
+  SNPProto.getFirstPaintFrame = function(options) {
+    if(window.requestAnimationFrame) {
+      var timeToFirstPaint = this.duration('navigationStart', 'firstPaintFrame', options);
```

Comment:
since I simplified the requirement for this to just return `NaN` if `firstPaintFrame` doesn't exist yet, you can make this `if` statement simply:
```js
    if(window.requestAnimationFrame) {
      return this.duration('navigationStart', 'firstPaintFrame', options);
    } else { ...
```